dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java

```diff
@@ -74,30 +80,70 @@ public EventResource(EventDAO eventDAO, UserDAO userDAO) {
+        }
+    }
+
     @GET
     @Timed
     @UnitOfWork(readOnly = true)
     @Operation(description = DESCRIPTION, summary = SUMMARY, security = @SecurityRequirement(name = "bearer"))
     @ApiOperation(value = SUMMARY, authorizations = {
         @Authorization(value = JWT_SECURITY_DEFINITION_NAME)}, notes = DESCRIPTION, responseContainer = "List", response = Event.class)
+    @ApiResponse(responseCode = HttpStatus.SC_OK + "", description = "A list of events", content = @Content(mediaType = MediaType.APPLICATION_JSON, array = @ArraySchema(schema = @Schema(implementation = Event.class))))
     public List<Event> getEvents(@Parameter(hidden = true) @ApiParam(hidden = true) @Auth User user, @QueryParam("event_search_type") EventSearchType eventSearchType, @Min(1) @Max(MAX_LIMIT) @DefaultValue(PAGINATION_DEFAULT_STRING) @ApiParam(defaultValue = PAGINATION_DEFAULT_STRING, allowableValues = PAGINATION_RANGE) @Parameter(schema = @Schema(maximum = "100", minimum = "1")) @QueryParam("limit") int limit, @QueryParam("offset") @DefaultValue("0") Integer offset) {
         User userWithSession = this.userDAO.findById(user.getId());
+        return getEventsForUser(userWithSession, eventSearchType, limit, offset);
+    }
+
+    @GET
+    @Timed
+    @UnitOfWork(readOnly = true)
+    @Path("/{userId}")
+    @Operation(description = "No authentication.", summary = "Get events based on filter and user id.")
+    @ApiOperation(value = "List recent events for a user.", notes = "No authentication.", response = Event.class, responseContainer = "List")
+    @ApiResponse(responseCode = HttpStatus.SC_OK + "", description = "A list of events", content = @Content(mediaType = MediaType.APPLICATION_JSON, array = @ArraySchema(schema = @Schema(implementation = Event.class))))
+    @ApiResponse(responseCode = HttpStatus.SC_NOT_FOUND + "", description = "User not found")
+    public List<Event> getUserEvents(@ApiParam(value = "User ID", required = true) @PathParam("userId") Long userId,
+        @QueryParam("event_search_type") EventSearchType eventSearchType,
+        @Min(1) @Max(MAX_LIMIT) @DefaultValue(PAGINATION_DEFAULT_STRING) @ApiParam(defaultValue = PAGINATION_DEFAULT_STRING, allowableValues = PAGINATION_RANGE) @Parameter(schema = @Schema(maximum = "100", minimum = "1")) @QueryParam("limit") int limit,
+        @QueryParam("offset") @DefaultValue("0") Integer offset) {
+        User user = this.userDAO.findById(userId);
+        checkUserExists(user);
+        return getEventsForUser(user, eventSearchType, limit, offset);
+    }
+
+    /**
+     * Returns events for the provided user
+     * @param user The user we are getting events for
+     * @param eventSearchType The types of events
+     * @param limit Event list limit
+     * @param offset Event list offest
+     * @return A list of events
+     */
+    private List<Event> getEventsForUser(User user, EventSearchType eventSearchType, int limit, Integer offset) {
+        if (eventSearchType == null) {
```

Comment:
I would remove this null check. If a null eventSearchType is not allowed, make it required in OpenAPI/Swagger doc, and throw a bad request from the resource handler up front, i.e., move the null check up there.

If you prefer to return an empty list, I would still make it required in OpenAPI/Swagger.

Reply:
Added `@NotNull` to the query parameter definition: https://github.com/dockstore/dockstore/pull/4901/commits/e842ae66b3809f99154104f1818c43d986489279 and removed the null check entirely.

Tested this with `curl`s, and it appears we don't need the explicit null check on our end, the following cases are handled based on our swagger/openapi definitions:
```
curl -X GET "http://localhost:4200/api/events/47091?eventSearchType=" -H "accept: application/json"

{"errors":["query param eventSearchType must not be null"]}
```
```
curl -X GET "http://localhost:4200/api/events/47091?eventSearchType=fakeParam" -H "accept: application/json"

{"code":400,"message":"query param eventSearchType must be one of [STARRED_ENTRIES, STARRED_ORGANIZATION, ALL_STARRED]"}
```
```
curl -X GET "http://localhost:4200/api/events/47091" -H "accept: application/json"

{"errors":["query param eventSearchType must not be null"]}
```
The above validations happen before making it to our function code for the API call (Maybe you already know this, but being verbose for my own edification).

Reply:
> The above validations happen before making it to our function code for the API call, though it's a bit odd the null parameter check is returned as an error rather than with a status code... (Maybe you already know this, but being verbose for my own edification).

I did not know, thanks for clarifying. Does the `{"errors":["query param eventSearchType must not be null"]}` case return a 4xx status code? Probably...